spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/config/GatewayAutoConfiguration.java

```diff
@@ -302,22 +303,23 @@ public GRPCResponseHeadersFilter gRPCResponseHeadersFilter() {
 	@Bean
 	@ConditionalOnEnabledFilter
 	@ConditionalOnProperty(name = "server.http2.enabled", matchIfMissing = true)
-	@ConditionalOnClass(name = "io.grpc.Channel")
-	public JsonToGrpcGatewayFilterFactory jsonToGRPCFilterFactory(GRPCSSLContext gRPCSSLContext,
+	@ConditionalOnClass(Channel.class)
+	public JsonToGrpcGatewayFilterFactory jsonToGRPCFilterFactory(GrpcSslConfigurer gRPCSSLContext,
 			ResourceLoader resourceLoader) {
 		return new JsonToGrpcGatewayFilterFactory(gRPCSSLContext, resourceLoader);
 	}
 
 	@Bean
 	@ConditionalOnEnabledFilter(JsonToGrpcGatewayFilterFactory.class)
-	@ConditionalOnMissingBean(GRPCSSLContext.class)
-	@ConditionalOnClass(name = "io.grpc.Channel")
-	public GRPCSSLContext gRPCSSLContext() throws KeyStoreException, NoSuchAlgorithmException {
+	@ConditionalOnMissingBean(GrpcSslConfigurer.class)
+	@ConditionalOnClass(Channel.class)
```

Comment:
This needs to remain `@ConditionalOnClass(name = "io.grpc.Channel")` since it is on a method and not a class.

Reply:
TIL! 🙇